contentcuration/contentcuration/static/js/edit_channel/models.js

```diff
@@ -1,63 +1,284 @@
 var Backbone = require("backbone");
 var _= require("underscore");
 
-window.ChannelEditDataModel = Backbone.Model.extend({
-	urlRoot: '/channel_edit',
-	initialize: function() {
-		//_.bindAll(this);
-	}
-});
-
-/* TODO: CHANGE TO ChannelEditDataModel.EXTEND */
-var TopicNodeModel = Backbone.Model.extend({
+var NodeModel = Backbone.Model.extend({
 	defaults: {
-        color1:"",
-		color2:"",
-		color3:"",
 		title:"Untitled",
-		description:"This is the description generated by default"
-    }
+		description:"No description",
+		parent: null,
+		children:[],
+    },
+    urlRoot: function() {
+		return window.Urls["node-list"]();
+	},
+	toJSON: function() {
+	  var json = Backbone.Model.prototype.toJSON.apply(this, arguments);
+	  json.cid = this.cid;
+	  return json;
+	},
+	parse: function(response) {
+	    return _(response).isArray()
+	         ? response[0]
+	         : response;
+	},
+
+	/*Used when copying items to clipboard*/
+    duplicate: function(parent_id){
+    	var title = this.generate_title(this.get("title"));
+		var data = {
+			title: title,
+			created : this.get("created"),
+			modified : this.get("modified"),
+			description: this.get("description"),
+			deleted: this.get("deleted"),
+			sort_order : this.get("sort_order"),
+			license_owner : this.get("license_owner"),
+			license: this.get("license"),
+			kind: this.get("kind"),
+			parent: parent_id
+		};
+		var node_data = new NodeModel(data);
+		var self = this;
+		node_data.save(data, {async:false,
+			success:function(){
+				self.copy_children(node_data, self.get("children"));
+			}
+		});
+		return node_data;
+	},
+
+	/* Function in case want to append (Copy #) to end of copied content*/
+	generate_title:function(title){
+    	var list = this.attributes.title.split(" ");
```

Comment:
I think this could be done a bit more neatly with a RegEx.

`
var matching = /\(Copy\s*([0-9]*)\)/g;
if (matching.test(title)) {
    title = title.replace(matching, function(match, p1) {
        // Already has "(Copy)"  or "(Copy <p1>)" in the title, so return either
        // "(Copy 2)" or "(Copy <p1+1>)"
        return "(Copy " + (p1==="" ? 2: Number(p1) + 1) + ")";
    });
}
`
Your 'else' statement can remain the same, but no need for the else if.